### PR TITLE
Added functionality for closing the app on absence of primary services

### LINKED ISSUE #205: Proccess Handler Bug
**Describe the bug**
When we initialize the database services on the `root.go` file, like this:

```golang
handler.Add("TransactionService", services.TransactionService.Run, services.TransactionService.Close)
handler.Add("BlockService", services.BlockService.Run, services.BlockService.Close)
```
the application doesn't stop because it can't ever reach the `handler.Close()` line because they are stuck on the`select{}` line on the `handler.Run()`.

**To Reproduce**
1. Go to `cmd/juno/cli/root.go`
2. Paste this code on line 59-60 (after RPC initialization and before `log.Default.Info("Starting all processes...")` line)

```golang
handler.Add("TransactionService", services.TransactionService.Run, services.TransactionService.Close)
handler.Add("BlockService", services.BlockService.Run, services.BlockService.Close)
```
3. Make sure the RPC is disabled on the configuration file (we only want the database services active).
4. Run Juno with `make all` command
5. You may see something like this:
```
2022-06-08T09:22:32.988-0400    INFO    cli/root.go:109 Using config file:      {"File": "/path/to/juno/juno.yaml"}
2022-06-08T09:22:32.988-0400    INFO    cli/root.go:127 Config values.  {"Database Path": "/path/to/juno", "Rpc Port": 8080, "Rpc Enabled": false}
2022-06-08T09:22:32.988-0400    INFO    cli/root.go:64  Starting all processes...
2022-06-08T09:22:32.988-0400    INFO    Block Service   services/service.go:34  Service started
2022-06-08T09:22:32.989-0400    INFO    Transaction Service     services/service.go:34  Service started
```
6. After that the only way to close the app is with a SIGTERM or SIGKILL

**Expected behavior**
I expect the application to exit immediately because the database services are "internal" services, so they cannot block the application if any other service, other than a primary one (such as RPC), is not running.

### PR DESCRIPTION
**Pull request by smishy05**
Resolves #205 

## Changes:
- Added a `PrimaryServiceCheck` to `root.go` which checks if primary services are running or not
- Added the condition for checking running of the service while closing it. If the app is not running and it is closed, then it leads to `nil reference` error.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by D-DePablos** [comment]
Could you remove the pb file changes?

File: internal/process/process.go [hunk 1]
```diff
@@ -89,3 +91,17 @@ func (h *Handler) Run() {
...
+func (h *Handler) PrimaryServiceChecker() int {
+	// notest
+	primarypkg := 0
+	for _, proc := range h.subprocs {
+		if proc.important {
+			primarypkg = primarypkg + 1
+		}
+	}
+	if primarypkg == 0 {
+		log.Default.Info("No primary processes running. Closing the app.")
```
Comment (D-DePablos) [inline internal/process/process.go:104]: good explanation msg

**Review by D-DePablos — CHANGES_REQUESTED** [review]
Only thing is to remove changes to protobuf files (state.pb, block.pb, abi.pb, transaction.pb)

**Comment by smishy05** [comment]
> Only thing is to remove changes to protobuf files (state.pb, block.pb, abi.pb, transaction.pb)

Done in the new commit. Please check

**Pull request merged**